hs/Database/CourseQueries.hs

```diff
@@ -127,44 +127,52 @@ buildSession lecs tuts =
 
 -- ** Other queries
 
--- | Gets Shape, Text and Path elements for rendering graph returned as JSON
-getGraphJSON :: Int64 -> IO Response
-getGraphJSON gId =
+-- | Looks up a graph using its title then gets the Shape, Text and Path elements
+-- for rendering graph (returned as JSON).
+getGraphJSON :: String -> IO Response
+getGraphJSON graphName =
     runSqlite databasePath $ do
-        sqlTexts    :: [Entity Text] <- selectList [TextGraph ==. toSqlKey gId] []
-        sqlRects    :: [Entity Shape] <- selectList
-                                             [ShapeType_ <-. [Node, Hybrid],
-                                              ShapeGraph ==. toSqlKey gId] []
-        sqlEllipses :: [Entity Shape] <- selectList
-                                             [ShapeType_ ==. BoolNode,
-                                              ShapeGraph ==. toSqlKey gId] []
-        sqlPaths    :: [Entity Path] <- selectList [PathGraph ==. toSqlKey gId] []
-        
-        let             
-            keyAsInt :: PersistEntity a => Entity a -> Integer
-            keyAsInt = fromIntegral . (\(PersistInt64 x) -> x) . head . keyToValues . entityKey
-            
-            texts          = map entityVal sqlTexts
-            rects          = zipWith (buildRect texts)
-                                     (map entityVal sqlRects)
-                                     (map keyAsInt sqlRects)
-            ellipses       = zipWith (buildEllipses texts)
-                                     (map entityVal sqlEllipses)
-                                     (map keyAsInt sqlEllipses)
-            paths          = zipWith (buildPath rects ellipses)
-                                     (map entityVal sqlPaths)
-                                     (map keyAsInt sqlPaths)
-            regions        = filter pathIsRegion paths
-            edges          = filter (not . pathIsRegion) paths
-            regionTexts    = filter (not .
-                                     intersectsWithShape (rects ++ ellipses))
-                                    texts
-                                    
-            result = createJSONResponse ["texts" .= (texts ++ regionTexts), 
-                                         "shapes" .= (rects ++ ellipses), 
-                                         "paths" .= (paths ++ regions ++ edges)]
-                                         
-        return result
+        graphEnt :: (Maybe (Entity Graph)) <- selectFirst [GraphTitle ==. graphName] []
+        case graphEnt of
+                Nothing -> return $ createJSONResponse ["texts" .= (),
```

Comment:
1. The indentation for this entire block is off. Remember to use 4 spaces, and no tabs.
2. Don't use `()`. You can use `[]` if this is supposed to be a list of texts, which I think it is.